velox/experimental/cudf/exec/CudfHashAggregation.cpp

```diff
@@ -507,8 +553,15 @@ auto toIntermediateAggregators(
     auto const inputIndex = aggregationNode.groupingKeys().size() + i;
     auto const kind = aggregate.call->name();
     auto const constant = nullptr;
-    aggregators.push_back(
-        createAggregator(step, kind, inputIndex, constant, isGlobal));
+    std::vector<TypePtr> argumentTypes;
+    for (auto const& arg : aggregate.call->inputs()) {
+      argumentTypes.push_back(arg->type());
+    }
+    const auto originalName = getOriginalName(kind);
+    const auto resultType =
+        exec::Aggregate::finalType(originalName, argumentTypes);
```

Comment:
the step here is always `kIntermediate` so should this be `exec::Aggregate::intermediateType`?